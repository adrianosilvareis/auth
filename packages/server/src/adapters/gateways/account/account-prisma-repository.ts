// import prisma from "@/frameworks/database/prisma-client-helper"
import prisma from "@/frameworks/database/prisma-client-helper"
import { uuid, Accounts, AccountModel, AccountStatusEnum } from "@auth/entity"
import { AccountsByGroupRepository } from "@auth/use-case"

export class AccountPrismaRepository implements AccountsByGroupRepository {
  async getAccountByGroup(authGroupId: uuid): Promise<Accounts[]> {
    const list = await prisma.as_auth_groups_authentications.findMany({
      where: {
        auth_groups_id: authGroupId
      },
      include: {
        as_authentications: {
          include: {
            as_accounts: true
          }
        }
      }
    })

    return list.map(({ as_authentications }) => {
      const { as_accounts } = as_authentications
      const account = {
        id: as_accounts.id,
        name: as_accounts.name,
        username: as_accounts.username,
        email: as_accounts.email,
        phone: as_accounts.phone,
        active: as_accounts.active,
        status: AccountStatusEnum[as_accounts.status],
        createdAt: new Date(as_accounts.created_at)
      }

      return new AccountModel(account)
    })
  }
}
