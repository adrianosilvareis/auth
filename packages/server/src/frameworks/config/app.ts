import express from 'express'

import { routes as setupRoutes } from './routes'

export const app = express()

setupRoutes(app)
