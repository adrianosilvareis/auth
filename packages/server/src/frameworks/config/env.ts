let DB_NAME, DB_HOST
const NODE_ENV = process.env.NODE_ENV ?? 'development'

if (NODE_ENV === 'production') { DB_NAME = 'clean_node_api' } else if (NODE_ENV === 'development') { DB_NAME = 'cna_dev' } else { DB_NAME = 'cna_test' }
if (NODE_ENV === 'test') { DB_HOST = 'localhost' } else { DB_HOST = 'postgres' }

export const env = {
  mongoUrl: process.env.MONGO_URL ?? 'mongodb://mongo:27017/clean-node-api',
  postgresUrl: process.env.DATABASE_URL ?? `postgres://postgres:postgres@${DB_HOST}:5432/${DB_NAME}`,
  port: process.env.PORT ?? 3000,
  jwtSecret: process.env.JWT_SECRET ?? 'a2611p@r19'
}
