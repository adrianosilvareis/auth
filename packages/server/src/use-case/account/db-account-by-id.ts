import { AccountModel, GetAccountById, uuid } from "@auth/entity";
import { GetAccountByIdRepository } from "@auth/use-case";

export class DbAccountById implements GetAccountById {
  constructor(private readonly accountRepo: GetAccountByIdRepository) {}

  async getById(accountId: uuid): Promise<AccountModel> {
    const account = await this.accountRepo.getAccountById(accountId);
    return account;
  }
}
