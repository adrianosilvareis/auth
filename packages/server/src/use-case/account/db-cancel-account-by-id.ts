import { Accounts, CancelAccountById, uuid } from "@auth/entity"
import { CancelAccountByIdRepository } from "@auth/use-case"

export class DbCancelAccountById implements CancelAccountById {
  constructor(private readonly accountRepo: CancelAccountByIdRepository) {}

  async cancel(accountId: uuid): Promise<Accounts> {
    const account = await this.accountRepo.cancel(accountId)
    return account
  }
}
