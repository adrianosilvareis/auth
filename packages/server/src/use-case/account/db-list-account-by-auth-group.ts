import { Accounts, ListAccountByAuthGroup, uuid } from "@auth/entity"
import { AccountsByGroupRepository } from "@auth/use-case"

export class DbListAccountByAuthGroup implements ListAccountByAuthGroup {
  constructor(private readonly accountRepo: AccountsByGroupRepository) {}

  async listAccountByAuthGroupId(authGroupId: uuid): Promise<Accounts[]> {
    const list = await this.accountRepo.getAccountByGroup(authGroupId)
    return list
  }
}
