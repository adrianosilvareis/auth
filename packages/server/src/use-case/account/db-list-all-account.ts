import { AccountModel, ListAllAccount } from "@auth/entity";
import { ListAllAccountRepository } from "@auth/use-case";

export class DbListAllAccount implements ListAllAccount {
  constructor(private readonly accountRepo: ListAllAccountRepository) {}

  async listAll(): Promise<AccountModel[]> {
    const response = await this.accountRepo.listAll();
    return response;
  }
}
