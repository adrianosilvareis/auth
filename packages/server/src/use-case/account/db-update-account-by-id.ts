import { AccountModel, UpdateAccountById, uuid } from "@auth/domain";
import { UpdateAccountByIdRepository } from "./protocols/update-account-by-id-repository";

export class DbUpdateAccountById implements UpdateAccountById {
  constructor(private readonly accountRepo: UpdateAccountByIdRepository) {}

  async update(
    accountId: uuid,
    data: Partial<AccountModel>
  ): Promise<AccountModel> {
    const updated = await this.accountRepo.update(accountId, data);
    return updated;
  }
}
