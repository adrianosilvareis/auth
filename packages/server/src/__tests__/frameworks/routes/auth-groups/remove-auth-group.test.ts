import request from "supertest"
import { app } from "@/frameworks/config"
import { prismaMock } from "@/__tests__/frameworks/database/prisma-client-mock"
import { as_auth_groups } from "@prisma/client"
import { mocked_authentication__group } from "@/__tests__/entity/mock/account"

describe("DELETE /api/auth-groups/:id", () => {
  it("should return HttpResponse serverError on repository throws", async () => {
    // @ts-ignore
    prismaMock.as_auth_groups_authentications.findMany.mockReturnValue([])
    // @ts-ignore
    prismaMock.as_auth_groups.delete.mockRejectedValueOnce(
      new Error("any_repository_error")
    )

    const { body, statusCode } = await request(app).delete(
      "/api/auth-groups/any_id"
    )

    expect(statusCode).toBe(500)
    expect(body).toMatchSnapshot()
  })
  it("should return HttpResponse badRequest on authGroup used", async () => {
    // @ts-ignore
    prismaMock.as_auth_groups_authentications.findMany.mockReturnValueOnce([
      mocked_authentication__group
    ])
    // @ts-ignore
    prismaMock.as_auth_groups.delete.mockRejectedValueOnce(
      new Error("any_repository_error")
    )

    const { body, statusCode } = await request(app).delete(
      "/api/auth-groups/any_id"
    )

    expect(statusCode).toBe(500)
    expect(body).toMatchSnapshot()
  })
  it("should return HttpResponse Ok on success", async () => {
    mockAll()

    const { body, statusCode } = await request(app).delete(
      "/api/auth-groups/any_id"
    )

    expect(statusCode).toBe(200)
    expect(body).toMatchSnapshot()
  })
})

const mockAuthGroupPrismaRepository: as_auth_groups = {
  id: "37f2e75e-4b86-4a99-977e-2aea6d4c782b",
  title: "Administrator"
}

function mockAll(): void {
  /** @ts-ignore */
  prismaMock.as_auth_groups_authentications.findMany.mockReturnValue([])

  /** @ts-ignore */
  prismaMock.as_auth_groups.delete.mockReturnValue(
    /** @ts-ignore */
    mockAuthGroupPrismaRepository
  )
}
