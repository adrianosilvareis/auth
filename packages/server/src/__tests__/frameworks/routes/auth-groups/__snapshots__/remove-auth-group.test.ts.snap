// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DELETE /api/auth-groups/:id should return HttpResponse Ok on success 1`] = `
Object {
  "id": "37f2e75e-4b86-4a99-977e-2aea6d4c782b",
  "title": "Administrator",
}
`;

exports[`DELETE /api/auth-groups/:id should return HttpResponse badRequest on authGroup used 1`] = `
Object {
  "error": Object {
    "message": "Internal server error",
    "stack": "Error: auth group in use
    at DbRemoveAuthGroup.remove (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\packages\\\\server\\\\src\\\\use-case\\\\auth-group\\\\db-remove-auth-group.ts:16:13)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at RemoveAuthGroupController.handler (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\packages\\\\server\\\\src\\\\adapters\\\\controllers\\\\auth-group\\\\remove-auth-group-controller.ts:14:24)
    at D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\packages\\\\server\\\\src\\\\frameworks\\\\adapters\\\\express-router-adapter.ts:11:26",
  },
}
`;

exports[`DELETE /api/auth-groups/:id should return HttpResponse serverError on repository throws 1`] = `
Object {
  "error": Object {
    "message": "Internal server error",
    "stack": "Error: any_repository_error
    at Object.<anonymous> (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\packages\\\\server\\\\src\\\\__tests__\\\\frameworks\\\\routes\\\\auth-groups\\\\remove-auth-group.test.ts:13:7)
    at Object.asyncJestTest (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\jasmineAsyncInstall.js:106:37)
    at D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queueRunner.js:45:12
    at new Promise (<anonymous>)
    at mapper (D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queueRunner.js:28:19)
    at D:\\\\BIBLIOTECAS\\\\Desktop\\\\auth-monorepo\\\\node_modules\\\\@jest\\\\core\\\\node_modules\\\\jest-jasmine2\\\\build\\\\queueRunner.js:75:41
    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
  },
}
`;
